app/demo-geant-integration/HitRootIO.cc

```diff
@@ -0,0 +1,205 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file demo-geant-integration/HitRootIO.cc
+//---------------------------------------------------------------------------//
+#include "HitRootIO.hh"
+
+#include <cstdio>
+#include <regex>
+#include <G4Event.hh>
+#include <G4RunManager.hh>
+#include <G4Threading.hh>
+#include <TBranch.h>
+#include <TFile.h>
+#include <TObject.h>
+#include <TROOT.h>
+#include <TTree.h>
+
+#include "corecel/Macros.hh"
+#include "corecel/io/Logger.hh"
+#include "accel/ExceptionConverter.hh"
+#include "accel/SetupOptions.hh"
+
+#include "GlobalSetup.hh"
+#include "SensitiveHit.hh"
+
+namespace demo_geant
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Create a ROOT output file for each worker except the master thread if MT
+ */
+HitRootIO::HitRootIO()
+{
+    ROOT::EnableThreadSafety();
+
+    file_name_ = std::regex_replace(
+        GlobalSetup::Instance()->GetSetupOptions()->output_file,
+        std::regex("\\.json$"),
+        ".root");
+
+    if (G4Threading::IsWorkerThread())
+    {
+        file_name_ += std::to_string(G4Threading::G4GetThreadId());
+    }
+
+    if (G4Threading::IsWorkerThread()
+        || !G4Threading::IsMultithreadedApplication())
+    {
+        file_.reset(TFile::Open(file_name_.c_str(), "recreate"));
+        CELER_VALIDATE(file_->IsOpen(), << "Failed to open " << file_name_);
+        tree_.reset(new TTree(
+            "Events", "Hit collections", this->SplitLevel(), file_.get()));
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the static thread local singleton instance
+ */
+HitRootIO* HitRootIO::GetInstance()
+{
+    static G4ThreadLocalSingleton<HitRootIO> instance;
+    return instance.Instance();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Write sensitive hits to output in the form of HitRootEvent.
+ */
+void HitRootIO::WriteHits(G4Event const* event)
+{
+    G4HCofThisEvent* HCE = event->GetHCofThisEvent();
+    if (HCE == nullptr)
+    {
+        return;
+    }
+
+    // Write the collection of sensitive hits into HitRootEvent
+    auto hit_event = std::make_unique<HitRootEvent>();
+    hit_event->SetEventID(event->GetEventID());
+    HitRootEvent::HitContainer* hcmap = hit_event->GetHCMap();
+    for (int i = 0; i < HCE->GetNumberOfCollections(); i++)
+    {
+        G4VHitsCollection* hc = HCE->GetHC(i);
+        std::string hcname = hc->GetName();
+        {
```

Comment:
What's the extra block for here?

Reply:
Removed  - leftover when selecting a specific collection name.